framework/src/components/storage/entities/account.js

```diff
@@ -381,12 +383,27 @@ class Account extends BaseEntity {
 			parsedFilters,
 		};
 
-		return this.adapter.executeFile(
-			parsedOptions.extended ? this.SQLs.selectFull : this.SQLs.selectSimple,
-			params,
-			{ expectedResultCount },
-			tx
-		);
+		return this.adapter
+			.executeFile(
+				parsedOptions.extended ? this.SQLs.selectFull : this.SQLs.selectSimple,
+				params,
+				{ expectedResultCount },
+				tx
+			)
+			.then(resp => {
+				const parseResponse = account => {
+					if (parsedOptions.extended) {
+						account.asset = account.asset ? account.asset : {};
+					}
+					return account;
+				};
+
+				if (expectedResultCount === 1) {
+					return parseResponse(resp);
+				}
+
+				return resp.map(parseResponse);
+			});
```

Comment:
You don't need all this custom code here, its too expensive and unnecessary to map over hundred of result set. You already added `asset` only to `get_extended.sql` file. So if someone called in other case there will be no `asset` property in response and that's what intended.